### ISSUE TITLE
Spamming for google servers when external network has no internet connection

### ISSUE THREAD

**Issue by backsy** [radioman/greatmaps#102]
Hi, i have a setup where i have a mapproxy server delivering the tiles and greatmaps showing the maps. As long as the network interfaces are not connected to any network everything works lightning fast. However when i connect to one of my networks that dont have internet access the speed of getting the tiles slows down to a trickle. I initially thought it was to do with mapproxy and the that maybe windows was the culprit but it now appears to be greatmaps causing the issue.

The mapproxy server that i have run on localhost to the consumer of the maps (i have 100gb maps and trying to chache it in greatmaps didnt seem to work properly - everything was uber slow). As such no request should reach out to the connected network. Instead what i see with wireshark is that there is alot of traffic looking for mt0.google.com (might be nt0, dont remember) and other similar servers.

It seems to me that even tho the server to ask tiles from is on localhost some external server is polled with no result as there is no internet connection available and as a result the tiles from the local server rarely get asked.

Is this just something that i have configured wrong for this behaviour to happen or might this be a bug?

TL:DR
Mapproxy server on localhost delivers wms tiles. If no external connection - map gets downloaded fine. If i have internet connection - map gets downloaded fine. If i have external connection but no internet - google servers are polled and localhost server gets nearly no tile requests.

**Comment by radioman** [comment]
i'm confused, can you simplify?  ;/

**Comment by backsy** [comment]
There seems to be some issue where if there is no internet connection and no adapters connected the external server gets polled for tiles fine and when there is a connection everything is fine but when there is a connection on an adapter but no internet access in that network the server does not get polled.

There is a server running on localhost but when i have connected to a wifi network that has no internet connection i see that instead of asking for tiles on the localhost address instead mt0.google.com and mt2.google.com connection requests are sent out.
